File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int min = Integer.MAX_VALUE;
        
        for(int i=0;i<prices.length;i++) {
            if(min > prices[i]){
                min = prices[i];
            }
            if(prices[i] - min > maxProfit){
                maxProfit = prices[i] - min;
            }
        }
        return maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    var maxProfit = 0
    var min = math.MaxInt64;
    
    for i:=0;i<len(prices);i++ {
        if min > prices[i] {
            min = prices[i]
        }
        if prices[i] - min > maxProfit {
            maxProfit = prices[i] - min
        }
    }
    
    return maxProfit
}